### PR TITLE
Yixiao header unittest

### PR DESCRIPTION
**Pull request by JYXiao-2021**
# Description
Create the unit test for the Header component to confirm whether it working as intended.

## Main changes explained:
- Create file Header.test.js to do the unit test.

## How to test:
1.check into current branch
2.do npm install to run this PR locally
3.run npm test src/components/Header/`__tests__`/Header.test.js

## Screenshots or videos of changes:
changed version:
image

### CODE REVIEW

File: src/components/Header/__tests__/Header.test.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import axios from 'axios';
+import { MemoryRouter as Router } from 'react-router-dom';
+import { Header } from '../Header'; 
+
+jest.mock('axios');
+const mockStore = configureMockStore();
+
+describe('Header Component', () => {
+  let store;
+  const mockStore = configureMockStore([]);
```
Comment (mianmiantea2019) [inline src/components/Header/__tests__/Header.test.js:14]: 'mockStore' is already declared in the upper scope on line 10

**Review by chappdev2019 — CHANGES_REQUESTED** [review]
Hi Yixiao. All the tests have passed! There was just one minor issue: I noticed the mockStore declaration was duplicated, but overall, great work! 

image

**Review by aaryaneil — CHANGES_REQUESTED** [review]
mockStore declared twice otherwise looks good to me.
Screenshot 2024-03-29 at 6 50 19 PM

**Review by linkhanthtel — DISMISSED** [review]
I reviewed your PR and all tests passed!

**Review by linkhanthtel — COMMENTED** [review]
One thing, 'mockStore' is declared twice in line 10 and 14

Screenshot 2024-03-30 at 2 34 03 PM

**Comment by JYXiao-2021** [comment]
@aaryaneil @mianmiantea2019 I did remove the duplicate code, please review again. Thank you.

**Review by scoopstroop92 — CHANGES_REQUESTED** [review]
Hello @JYXiao-2021 Good job on the existing test cases.  However I believe more test cases and validations can be added

**Review by vikrambadhan — DISMISSED** [review]
Hi, I tested your PR and all the test cases passed!

Week 31 - PR4

**Review by bhuvaneswarignanasekar — DISMISSED** [review]
I have reviewed your PR. Works as intented. 
All test cases passed. 
Screenshot 2024-04-09 at 8 17 53 PM

File: src/components/Header/__tests__/Header.test.js (hunk 1 above)
Comment (mianmiantea2019) [reply src/components/Header/__tests__/Header.test.js]: Good work and all tests passed!

**Review by chappdev2019 — DISMISSED** [review]
Good work and all tests passed!

**Review by jennZhang93 — DISMISSED** [review]
All unit test passed. good work!
image

**Review by yvonnep165 — DISMISSED** [review]
Hi @JYXiao-2021 , I checked all test cases and ran the tests. All tests passed. Nice work! Please refer to the screenshot below:
Screenshot 2024-04-26 at 5 07 05 PM

**Review by Tiantian-C — DISMISSED** [review]
All tests have been passed!
Screenshot 2024-04-27 at 6 45 31 PM

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**